### ISSUE TITLE
Using the binary path makes the databases less portable

### ISSUE THREAD

**Issue by ikelos** [angr/binsync#175]
At the moment, the binary path is extracted from the database in question and that path is used to keep the persistent storage.  This creates issues if the database is shared, but not the original binary (as can happen with IDA, for example).  This is more noticeable where it may be impracticable to create similar directory structures (such as from windows to mac).  This will cause a failure when loading a project:

```
INFO | 2022-08-09 12:00:59,583 | binsync.common.ui.config_dialog | Client has connected to sync repo with user: ####.
Traceback (most recent call last):
  File "/tmp/binsync/binsync/common/ui/config_dialog.py", line 184, in _on_ok_clicked
    saved_config = self.save_config()
  File "tmp/binsync/binsync/common/ui/config_dialog.py", line 245, in save_config
    return config.save()
  File "/tmp/binsync/binsync/data/configuration.py", line 21, in save
    with open(self.path, "w") as fp:
FileNotFoundError: [Errno 2] No such file or directory: <windows-path-from-ida-file>
```

I understand the reason for tying it to the binary, since this is common across all the backends (angr, binja, IDA, etc).  Unfortunately it appears difficult to override since the persistent project settings *are* what's being stored in that path (which fails) so there's no way to tell the project to use a different path before it looks there (even if you provide full details in the configuration dialog).

I'd suggest error verification for attempts to read/write to paths that don't exist and also that project persistent configurations are written somewhere reliable, either into the repo itself, or where the database that's been opened lives (rather than where it says its source binary lived).  It may also be possible to check for the file existence and return None if it's not present, in ida's `compat.get_binary_path()` but I don't know where else that method is used, so it might have unintended consequences?  I'm not sure which of those are feasible, but happy to test any suggestions...  5:)

https://github.com/angr/binsync/pull/86/files#diff-adf5477db2fb5dc1210d5a783d5fed8fd8cba544eee5f8c0123d098b8bab8f3cR420

**Comment by mahaloz** [comment]
@ikelos Ah yes, I as this issue ones in defcon quals. Thanks for making an issue for it I forgot to document that it happened so someone. 

The fundamental problem here is our reliance on the binary, which I agree is not ideal. Even in ghidra and binja, you can use their database file without having the original binary anymore... so we need a way around this.

Fortunately, I think I know a path to fixing this rather easily. At the end of the day, all we really need is the binary's hash to validate you have the same binary. Even if this does not work, we should not crash.

**Comment by mahaloz** [comment]
Forgot I should also give credit to @WGibbs15 for finding this bug in quals lol

**Comment by mahaloz** [comment]
@ikelos alright this bug should be fixed in master with the pr of #177, Try it then, I'll reopen this issue if it still does not work :)

**Comment by ikelos** [comment]
Yep, looks to have sorted the errors, thanks!  I'm not sure the configuration is persistent, I couldn't find a hidden `.conf` file in either the git repo or where the database is, but I can cope with that as long as it lets me get further with the syncing...  5:)
